eav/v1/ReadApi/Database/BsonQuery.cs

```diff
@@ -0,0 +1,119 @@
+﻿
+namespace ReadApi.Database
+{
+    using System;
+    using MongoDB.Bson;
+
+    public static class BsonQuery
+    {
+        /// <summary>
+        /// Creates a filter to query on documents that match the specified reference date
+        /// and have their 'is deleted' status set to false.
+        /// </summary>
+        /// <param name="referenceDate"></param>
+        /// <param name="filterInput"></param>
+        /// <param name="filterOnParentDeletedStatus"></param>
+        /// <returns>A <see cref="BsonDocument"/> that can be used to filter documents.</returns>
+        public static BsonDocument CreateMutationsFilter(DateTime referenceDate, string filterInput = "$Mutations",
+            bool filterOnParentDeletedStatus = false)
+        {
+            var bsonDocument = new BsonDocument
+            {
+                {
+                    "$filter", new BsonDocument
+                    {
+                        { "input", filterInput },
+                        { "as", "mutation" },
+                        {
+                            "cond", new BsonDocument
+                            {
+                                {
+                                    "$and", new BsonArray
+                                    {
+                                        new BsonDocument
+                                        {
+                                            {
+                                                "$eq", new BsonArray
+                                                {
+                                                    "$$mutation.IsDeleted", false
+                                                }
+                                            }
+                                        },
+                                        new BsonDocument
+                                        {
+                                            {
+                                                "$gte", new BsonArray
+                                                {
+                                                    referenceDate, "$$mutation.StartDate"
+                                                }
+                                            }
+                                        },
+                                        new BsonDocument
+                                        {
+                                            {
+                                                "$lte", new BsonArray
+                                                {
+                                                    referenceDate, "$$mutation.EndDate"
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            if (filterOnParentDeletedStatus)
```

Comment:
It's just a PoC, but I'm asking myself if wouldn't be better to have a method CreateMutationsFilterWithParent? then call this one passing the argument as true/false?